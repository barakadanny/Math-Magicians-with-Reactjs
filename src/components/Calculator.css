.calculator-container {
  display: flex;
  justify-content: space-between;
}

.calculator {
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  width: 400px;
  box-shadow: 0 3px 10px 3px #aaa;
  border: 0.01em solid #c4c4c4;
}

.result {
  width: 100%;
  background-color: #858693;
  min-height: 1em;
  color: white;
}

.result > p {
  padding: 0 0.2em;
  text-align: right;
  font-size: 2em;
  line-height: 0.1em;
}

span {
  box-sizing: border-box;
  text-align: center;
  padding: 1em;
  cursor: pointer;
  width: 25%;
  background-color: #e0e0e0;
  line-height: 2em;
  border-top: 0.01em solid #c4c4c4;
  border-left: 0.01em solid #c4c4c4;
}

.operator {
  background-color: #ff9500;
}

.double {
  width: 50%;
}

.clicked {
  background-color: #cc8400;
}

/* mobile version */
@media (max-width: 600px) {
  .calculator-container {
    flex-direction: column;
  }

  .calculator-usage {
    margin-bottom: 1em;
    text-align: center;
  }
}
