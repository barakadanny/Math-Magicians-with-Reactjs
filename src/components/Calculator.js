import React from 'react';
import './Calculator.css';

class Calculator extends React.Component {
  render() {
    return (
      <div className='calculator'>
        <div className='result'>
          <p></p>
        </div>

        <span className='clear grey'>AC</span>
        <span className='grey'>+/-</span>
        <span className='grey'>%</span>
        <span className='operator'>รท</span>

        <span>7</span>
        <span>8</span>
        <span>9</span>
        <span className='operator'>x</span>

        <span>4</span>
        <span>5</span>
        <span>6</span>
        <span className='operator'>-</span>

        <span>1</span>
        <span>2</span>
        <span>3</span>
        <span className='operator'>+</span>

        <span className='double'>0</span>
        <span>.</span>
        <span className='operator'>=</span>
      </div>
    );
  }
}

export default Calculator;
