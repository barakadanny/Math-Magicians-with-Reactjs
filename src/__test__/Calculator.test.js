import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';
import '@testing-library/jest-dom';

describe('Test Calculator pag', () => {
  it('testing Calculator page UI rendering', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
