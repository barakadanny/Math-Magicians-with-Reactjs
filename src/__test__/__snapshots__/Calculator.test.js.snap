// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Calculator pag testing Calculator page UI rendering 1`] = `
<div
  className="calculator-container"
>
  <div
    className="calculator-usage"
  >
    Lets do some Math!
  </div>
  <div
    className="calculator"
  >
    <div
      className="result"
    >
      <p>
        0
      </p>
    </div>
    <span
      onClick={[Function]}
      role="presentation"
    >
      AC
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      +/-
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      %
    </span>
    <span
      className="operator"
      onClick={[Function]}
      role="presentation"
    >
      รท
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      7
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      8
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      9
    </span>
    <span
      className="operator"
      onClick={[Function]}
      role="presentation"
    >
      x
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      4
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      5
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      6
    </span>
    <span
      className="operator"
      onClick={[Function]}
      role="presentation"
    >
      -
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      1
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      2
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      3
    </span>
    <span
      className="operator"
      onClick={[Function]}
      role="presentation"
    >
      +
    </span>
    <span
      className="double"
      onClick={[Function]}
      role="presentation"
    >
      0
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      .
    </span>
    <span
      className="operator"
      onClick={[Function]}
      role="presentation"
    >
      =
    </span>
  </div>
</div>
`;

exports[`Testing the Calculator page rendering`] = `
<div className="calculator-container">
      <div className="calculator-usage">Lets do some Math!</div>
      <div className="calculator">
        <div className="result">
          <p>{result || 0}</p>
        </div>

        <span onClick={updateObject} role="presentation">
          AC
        </span>
        <span onClick={updateObject} role="presentation">
          +/-
        </span>
        <span onClick={updateObject} role="presentation">
          %
        </span>
        <span onClick={updateObject} role="presentation" className="operator">
          รท
        </span>

        <span onClick={updateObject} role="presentation">
          7
        </span>
        <span onClick={updateObject} role="presentation">
          8
        </span>
        <span onClick={updateObject} role="presentation">
          9
        </span>
        <span onClick={updateObject} role="presentation" className="operator">
          x
        </span>

        <span onClick={updateObject} role="presentation">
          4
        </span>
        <span onClick={updateObject} role="presentation">
          5
        </span>
        <span onClick={updateObject} role="presentation">
          6
        </span>
        <span onClick={updateObject} role="presentation" className="operator">
          -
        </span>

        <span onClick={updateObject} role="presentation">
          1
        </span>
        <span onClick={updateObject} role="presentation">
          2
        </span>
        <span onClick={updateObject} role="presentation">
          3
        </span>
        <span onClick={updateObject} role="presentation" className="operator">
          +
        </span>

        <span onClick={updateObject} role="presentation" className="double">
          0
        </span>
        <span onClick={updateObject} role="presentation">
          .
        </span>
        <span onClick={updateObject} role="presentation" className="operator">
          =
        </span>
      </div>
    </div>
`;
